### DISCUSSION TITLE
Use PDO::FETCH_ASSOC

### DISCUSSION THREAD

**Discussion by zavre-c** [laravel/framework#39584] [Show and tell]
### USE PDO::FETCH_ASSOC

> For Example we will use sqlsrv here

In AppServiceProvider -> boot() , we replicated config in database.php with **name** _mssql_ and **driver** _mssql_

#### Use either of the both 

- Using DB::extend()

```php

DB::extend("mssql",function($config, $name) {
    $pdo = function() use ($config) {
        return (new SqlServerConnector)->connect($config);
    };
    return (new class($pdo,$config['database'],$config['prefix'],$config) extends SqlServerConnection {
        protected $fetchMode = PDO::FETCH_ASSOC;
    });
});

```

- Using DB::resolveFor() and Container Binding

```php

$this->app->bind("db.connector.mssql",function() {
    return new SqlServerConnector;
});
DB::resolverFor("mssql",function() {
    return (new class(...func_get_args()) extends SqlServerConnection {
        protected $fetchMode = PDO::FETCH_ASSOC;
    });
});

```

Tried this but didn't work [DEPRECATED]  #3512

`$this->app['config']['database.fetch'] = PDO::FETCH_ASSOC; `

**Comment by spudro228** [comment]
Laravel developers deliberately made the code not expandable
Weird thing
